src/shillelagh/adapters/base.py

```diff
@@ -51,10 +53,62 @@ class Adapter:
     # if true, the requested columns will be passed to ``get_rows`` and ``get_data``
     supports_requested_columns = False
 
+    # if true, a set of strings from an in statement will be serialized before being
+    # passed in as bounds to ``get_rows`` and ``get_data``. This requires
+    # supports_requested_columns to be set to True as well.
+    supports_in_statements = False
+
     def __init__(self, *args: Any, **kwargs: Any):  # pylint: disable=unused-argument
         # ensure ``self.close`` gets called before GC
         atexit.register(self.close)
 
+    @staticmethod
+    def serialize_set(a_set: Any) -> Any:
+        """Serialize the given set to a string.
+
+        This will convert a given set {'a', 'b', 'c'} to the string: "set(['a','b','c'])"
+
+        Args:
+            a_set (Any): A set (of strings) or None.
+
+        Raises:
+            TypeError: If anything but a set or None is given to this method.
+
+        Returns:
+            Any: A string representation of the set or None.
+        """
+        if a_set is None:
+            return None
+        if not isinstance(a_set, set):
+            raise TypeError("serialize_set(): a_set must be a set")
+        return f"set({json.dumps(list([item for item in a_set]))})"
```

Comment:
btw `list([item for item in a_set])` is the same as `list(item for item in a_set)` is the same as `list(a_set)`.

Reply:
I like the short version and for some reason I only have it in my fork. I'll commit it.